File: practice26/13.cpp

//https://www.nowcoder.com/practice/17a083854661490e85e5bb6c4b26e546?tpId=98&tqId=32966&tPage=8&rp=8&ru=/ta/2019test&qru=/ta/2019test/question-ranking
#include<iostream>
#include<vector>
using namespace std;

int main() {
	long hp;
	long normalattack;
	long bufferattack;
	while (cin >> hp >> normalattack >> bufferattack) {
		int x = 0;
		if (bufferattack <= normalattack * 2) {
			while (hp > 0) {
				hp = hp - normalattack;
				x++;
			}
		}
		else {
			while (hp > normalattack) {
				x = x + 2;
				hp = hp - bufferattack;
			}
			if (hp > 0) {
				x++;
			}
		}
		cout << x << endl;
	}
	system("pause");
	return 0;
}

File: practice33/13.java


//https://www.nowcoder.com/practice/14cf13771cd840849a402b848b5c1c93?tpId=85&tqId=29884&tPage=3&rp=3&ru=/ta/2017test&qru=/ta/2017test/question-ranking
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int num = in.nextInt();

        int[] m={1,5,10,20,50,100};
        long[] data=new long[num+1];
        for(int i=0;i<=num;i++){
            data[i]=1;
        }

        for(int i=1;i<6;i++){
            for(int n=1;n<=num;n++){
                if(n>=m[i]){
                    data[n]=data[n]+data[n-m[i]];
                }
            }
        }
        System.out.println(data[num]);
    }
}